hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueService.java

```diff
@@ -103,7 +104,7 @@
     private static final Object NULL_OBJECT = new Object();
 
     private final ConcurrentMap<String, QueueContainer> containerMap = new ConcurrentHashMap<>();
-    private final ConcurrentMap<String, LocalQueueStatsImpl> statsMap = new ConcurrentHashMap<>(1000);
+    private final ConcurrentMap<String, LocalQueueStatsImpl> statsMap = MapUtil.createConcurrentHashMap(10);
```

Comment:
Can't we initialize this with the number of DSs defined in the static config? Then we can grow on-demand if new ones created dynamically.

Reply:
yes, this was also an idea, have to see how (un)intrusive it is to get the config in the constructor and iterate through it.

Reply:
wildcard configs are a gotcha but I think it's fine to work with best-effort approach here.

Reply:
Addressed this comment, also on other places. The approach is pretty simple, I just take the number of configs, which means wildcard and default also count as one. PTAL